src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -221,25 +230,18 @@
   return-to-pool
   "Return a borrowed pool instance to its free pool."
   [instance :- jruby-schemas/JRubyInstanceOrPill]
-  (if (jruby-schemas/jruby-instance? instance)
```

Comment:
100% agreement. I already made the changes, but it looks like I forgot to commit them. That's a good point, I only found that this function had a problem once I ran the PS tests on it. I'll add a test in jruby-utils that exercises it